Communications (#33)
tilt-dev/company: communications/README.md

# Communications

## Async written first

- If you want to discuss an idea (e.g. new or existing product feature, update a marketing page), comment on or create a new story/epic in Clubhouse. @-mention specific individuals and ask them specific questions. See [Product Development](../product-development/README.md).
- Ensure your [Clubhouse notifications](https://help.clubhouse.io/hc/en-us/sections/201543416-Notifications) are properly configured. Respond to Tilters within a few days with an answer, or at least say you don't have time currently to do so.
- If you want to discuss a process change, consider the smallest reasonable change and create a GitHub PR in this repo as a proposal. @-mention and assign specific individuals for review.

> - Only when you have exhausted discussions in Clubhouse and/or GitHub, consider scheduling a meeting with a clear agenda to determine next steps.

Reviewer: lol, I think the phrase "exhausted discussions" is going to open some scar tissue for some people.

I'm worried that this point is going to be interpreted as "all discussions need to end in agreement, and if they don't end in agreement, you keep scheduling meetings until there is agreement". Which I don't think is that you're intending!

I might break this down into two different cases:

- the case where we're just doing initial ideation and brainstorming, but no work has started and no one has taken ownership of the problem yet

- the case where there is a DRI, and they're responsible for the next steps

and how/why each of those cases would evolve into a meeting

Author: @nicks : You are right, the purpose of this sentence was not to convey "getting to agreement". I actually wanted to address a simpler more generic case, which is to schedule a meeting when you need to, because it's getting to difficult to talk over async comments. So I think your two cases are actually _too_ specific, and not necessarily what I want to introduce here. What do you think of this?

```suggestion
- If you are having difficulty understanding each other over Clubhouse and/or GitHub, consider scheduling a meeting with a specific agenda to further discuss. But prefer starting with async communications first.
```

Reviewer: sgtm

> - When using Slack, prefer using public channels over DMs. @-mention specific individuals with a question if you want a response. Prefer using threads.

Reviewer: I might reframe this slightly. How would you feel about something like: 

> Suppose you have an idea for a feature, but you're not sure where it belongs. Or you see weird behavior in Tilt, and are not sure is a bug. Please ask in Slack! Ask in a shared channel, rather than a private DM.

> If you get confirmation that this is a common issue that people want to discuss more, move the more substantial discussion to a product-focused venue, like Github issues or a Clubhouse story

Author: I actually rather this _not_ be in Slack, and that people go to Clubhouse directly. I have lots of reasons for this, that not everyone might agree with now. So I'll just delete this line and we can re-visit again.

- Slack is an async tool. Don't expect people to respond immediately. Don't feel pressured to respond immediately.

> - Prefer Clubhouse over Slack, and Slack over email.

Reviewer: I might qualify this as "For product discussion, ..."

Author: I don't want to qualify it, because I think it applies more broadly. So I'd rather just delete it then. Will do that. Especially since the previous sentences already explain how to use Clubhouse.

# Communications

## Async written

- If you want to discuss an idea (e.g. new or existing product feature, update a marketing page), comment on or create a new story/epic in Clubhouse. @-mention specific individuals and ask them specific questions. See [Product Development](../product-development/README.md).

> - Ensure your [Clubhouse notifications](https://help.clubhouse.io/hc/en-us/sections/201543416-Notifications) are properly configured. Respond to Tilters within a few days with an answer, or at least say you don't have time currently to do so.

Reviewer 2: "a few days" seems like a really long time to me, especially when combined with instructions not to fall back to slack until Clubhouse has gotten in the way. I generally feel like I'm already often waiting longer than I'd like on Clubhouse replies. I don't know if I'm alone in this. It seems worth 1) polling the team on what we want to collectively commit to here, 2) adding a note that it's fine to ping people on slack for faster responses (or maybe that'd be similarly addressed by a "goals" bit at the top that says this is about establishing a source of truth rather than social stuff? I generally think a lot of process docs could greatly benefit from more discussion of the process's goals, but that's a longer conversation).

Author: Thanks @landism . What do you think about this?

```suggestion
- Ensure your [Clubhouse notifications](https://help.clubhouse.io/hc/en-us/sections/201543416-Notifications) are properly configured. Respond to Tilters as soon as you can with an answer, or at least say you don't have time currently to do so.
```

> I generally think a lot of process docs could greatly benefit from more discussion of the process's goals, but that's a longer conversation).

@landism : This is really close to something I asked @nicks about earlier. And there's a a few comments earlier in this PR between myself and @nicks that GitHub seems to have erased (😭) which mentions this. TLDR: I've asked Nick to establish some company values written down, which should ultimately drive company process. He is thinking about it so let this comment be another reminder to Nick, i.e. Matt is indirectly interesting it this too. (I think of company values and process goals to be really close. Processes should achieve certain process goals, which should be a reflection of more fundamental values.)

In the mean time, however, I don't that to slow down improving process. So the goal of this PR is to codify existing process we are already fairly aligned with. So I think there's some value to write down things we are already doing. And if we want to start moving toward formalizing more nuanced interactions or changing process, I'll continue to push for company values.

Reviewer 2: > I think of company values and process goals to be really close. Processes should achieve certain process goals, which should be a reflection of more fundamental values.

Hm, maybe you've got a much different experience with company values than I do. I've generally seen those as very high-level and abstract, with there still being value in discussing more proximate goals for specific processes. That seems like a bigger conversation and non-blocking for this PR, though, and maybe interesting for our next 1:1.

> So the goal of this PR is to codify existing process we are already fairly aligned with in order to set consistent expectations.

In this case, maybe we shouldn't talk about response time yet. I think we don't have a consistent expectation and response times vary greatly across the team. I personally feel like a few days is too long, but maybe most of the team doesn't, and that'd be fine. "a few days" to "as soon as possible" is a pretty significant difference when both are theoretically describing existing expectations! I actually don't know what our coworkers' existing expectations are. I was just saying that I personally think a few days is too long. I don't think my personal expectation should stand for the team's shared expectation.

I think it would be useful to come up with a time and document it here, but it's fine if you don't want to do that in this PR. I've found it very useful when we've put explicit times for exterminator response times in the working agreement. Setting it to a day or an hour really clarified how much we should worry about distractions and interrupting existing work.

As a bit of an aside, I think "as soon as possible" is not great for communicating expectations. It's rather subjective and context-sensitive. One obviously can't take it literally, so it becomes "as soon as is reasonably possible", and then some people might think a day is reasonable and other people might think two hours is unreasonable (I feel like we had a similar situation with the exterminator response time). We'll never take judgement out of the process, but objective units of time seem much more effective for aligning expectations.

Author: ```suggestion
- Ensure your [Clubhouse notifications](https://help.clubhouse.io/hc/en-us/sections/201543416-Notifications) are properly configured.
```

Author: Thanks @landism for raising those concerns. I just removed that sentence to resolve this PR. But very much I share your concerns and want to set these expectations properly in future PRs.

Author: @landism : GitHub seems to have fixed a weird bug. 🤷‍♂️ . Anyways here's the conversation I was talking about earlier: https://github.com/tilt-dev/company/pull/33#issuecomment-632211279.

> I've generally seen those as very high-level and abstract, with there still being value in discussing more proximate goals for specific processes.

I'm very interested in moving forward with something in this realm. So expect me to push for this in coming weeks and months. And we can figure out terms and definitions then.

- If you want to discuss a process change, consider the smallest reasonable change and create a GitHub PR in this repo as a proposal. @-mention and assign specific individuals for review.

> - If you are having difficulty understanding each other over Clubhouse and/or GitHub, consider scheduling a meeting with a specific agenda to further discuss. But prefer starting with async communications first.

Reviewer 2: I'd love something along the lines of "Afterwards, update Clubhouse/Github with any relevant output from the discussion, both to record the outcome and as a double-check on shared understanding."

Author: Thanks @landism. Definitely this is something I'd want to include, but probably for a more full-fledged meeting section in a later PR.

> - Slack is an async tool. Don't expect people to respond immediately. Don't feel pressured to respond immediately.

Reviewer 3: Do we expect that everyone will even SEE a (non-DM) communication in Slack? might be worth it to set an expectation around that.

Author: Thanks @maiamcc. I have more opinions on how Slack should be used. We can definitely collaborate on that in a future PR including what you mentioned. This PR is intended to be a small first step.
